### PR TITLE
Minor fixes related to the sequence number.

### LINKED ISSUE #354: Ruby client uses incorrect max sequence number?
My ruby is a little rusty, but I was looking at [this line](https://github.com/elasticsearch/logstash-forwarder/blob/master/lib/lumberjack/client.rb#L8), and when I pop this into `irb`, I get back:

    1.9.3-p385 :001 > SEQUENCE_MAX = (2**(0.size * 8 -2) -1)
     => 4611686018427387903

But, sequence numbers are 32 bit unsigned integers (according to [this](https://github.com/elasticsearch/logstash-forwarder/blob/master/PROTOCOL.md#data-frame-type)) so this can't possibly be correct.

### PR DESCRIPTION
**Pull request by purbon**
Couple of minor fixes:

* Made sure the Sequence size is of size 32 bits unsigned int. Fixes #354.
* Added a couple of test for the sequence numbers, one for the max value and another for the ack changes related to the windows size.

### CODE REVIEW

**Comment by jordansissel** [comment]
Hmm, specs don't pass for me in this PR, and I can't rebase it against master. Can you fix that?

**Comment by purbon** [comment]
on it

**Comment by purbon** [comment]
HI,
  rebased against master and run the specs:

```
➜  logstash-forwarder git:(fix/sequence_num) bundle exec rspec
...

Finished in 15.02 seconds (files took 1.26 seconds to load)
3 examples, 0 failures
```
they pass for me now, let me know if there is anything else I can do here.

File: .logstash-forwarder [hunk 1]
```diff
@@ -0,0 +1 @@
+{"/var/folders/dp/r65wtk6551lc5q27jjn89rtm0000gn/T/input_file-8a4d56172b8bfdcac2dae44712f7bf704559b456a8082e25800894f5049a":{"source":"/var/folders/dp/r65wtk6551lc5q27jjn89rtm0000gn/T/input_file-8a4d56172b8bfdcac2dae44712f7bf704559b456a8082e25800894f5049a","offset":55,"inode":12200034,"device":16777220}}
```
Comment (jordansissel) [inline .logstash-forwarder]: Oops, I think this got added by mistake. I'll prune it before merging.

**Comment by jordansissel** [comment]
Superseded by #372 (fork of this pr)

**Pull request closed without merging**